mbed/mbed.py

```diff
@@ -1652,8 +1672,14 @@ def check_requirements(self, require_install=False):
                     pkg = re.sub(r'-', '_', re.sub(r'^([^<>=@]+).*$', r'\1', line).lower())
                     pkg = re.sub(r'^(git|hg|svn|bzr)\+(.*)/([\w.-]+?)(\.(git|hg))?$', r'\3', pkg)
                     pkg = re.sub(r'\[([\w]+)\]', '', pkg)
-                    if not pkg in installed_packages:
-                        missing.append(pkg)
+                    line_parts = line.split(";")
+
+                    for line_part in line_parts[1:]:
```

Comment:
Why was the `1:` needed?

Reply:
A python requirements file looks like this:

```
pyusb>=1.0.0,<2.0.0
hidapi>=0.7.99,<0.8.0;platform_system!="Linux"
```

`line_parts` for the first line would look like: `["pyusb>=1.0.0,<2.0.0"]`.
For the second one, it would look like `["hidapi>=0.7.99,<0.8.0", "platform_system!=\"Linux\""]`.
So the `[1:]` only selects what comes after the first `;`, if there is one, which is the environment markers that we're trying to parse.